ocs_ci/ocs/ui/mcg_ui.py

```diff
@@ -248,16 +248,56 @@ def delete_bucketclass_ui(self, bc_name):
         self.do_click(self.generic_locators["confirm_action"])
 
 
-class ObcUI(PageNavigator):
+class BucketsUI(PageNavigator):
+    """
+    A class representation for abstraction of OBC or OB-related OpenShift UI actions
+
+    """
+
+    def __init__(self, driver):
+        super().__init__(driver)
+        self.ocs_version = str(version.get_ocs_version_from_csv(only_major_minor=True))
+        self.obc_loc = locators[self.ocs_version]["obc"]
+
+    def select_openshift_storage_default_project(self):
+        """
+        Helper function to select openshift-storage project
+
+        Notice: the func works from PersistantVolumeClaims, VolumeSnapshots and OBC pages
+        """
+        logger.info("Select openshift-storage project")
+        self.do_click(self.generic_locators["project_selector"])
+        self.wait_for_namespace_selection(
+            project_name=config.ENV_DATA["cluster_namespace"]
+        )
+
+    def delete_resource(self, delete_via, resource):
+        if delete_via == "Actions":
+            logger.info(f"Go to {resource} Page")
+            self.do_click(self.obc_loc["resource_name"])
+
+            logger.info(f"Click on '{delete_via}'")
+            self.do_click(self.generic_locators["actions"])
+        else:
+            logger.info(f"Click on '{delete_via}'")
+            self.do_click(self.generic_locators["three_dots"])
+
+        logger.info(f"Click on 'Delete {resource}'")
+        self.do_click(self.obc_loc["delete_obc"])
+
+        logger.info(f"Confirm {resource} Deletion")
+        self.do_click(self.generic_locators["confirm_action"])
+
+
+class ObcUI(BucketsUI):
     """
     A class representation for abstraction of OBC-related OpenShift UI actions
 
     """
 
     def __init__(self, driver):
         super().__init__(driver)
-        ocs_version = f"{version.get_ocs_version_from_csv(only_major_minor=True)}"
-        self.obc_loc = locators[ocs_version]["obc"]
+        self.obc_loc = locators[self.ocs_version]["obc"]
```

Comment:
If it's inside init, the test will fail when unsupported ocs/odf version is used rather than skipping the test.

Reply:
if the ObcUI was not instantiated, if the test was skipped with decorator how that may happen? 

If the test is already running for checkin functionality not related to ObcUi tester should place <if-else> block before creating ObcUI instance. In any case it is wrong to put this logic in ObcUi constructor.